app/src/main/java/net/squanchy/schedule/filterschedule/FilterSchedule.kt

```diff
@@ -0,0 +1,135 @@
+package net.squanchy.schedule.filterschedule
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.os.Bundle
+import android.support.annotation.ColorInt
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.CheckBox
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_track_filters.*
+import net.squanchy.R
+import net.squanchy.schedule.domain.view.Track
+import net.squanchy.service.repository.TracksRepository
+
+class FilterScheduleActivity : AppCompatActivity() {
```

Comment:
This is basically a rewrite of what was in the `Dialog`. It's a full activity as I plan on making it a bottom sheet, or something along those lines; the goal is to have it fully portable to something totally encapsulated.

This activity observes the whole tracks list, coming from the DB, and the list of selected tracks that comes from the `InMemoryTracksFilter` (initially they have the same contents). These are combined in `CheckableTrack`s, a typealias for `Pair<Track, Boolean>`.

The `List<CheckableTracks>` is then fed to the `TrackFiltersAdapter` (naming to be improved!) which uses it as a view model for its items. When one of the items is tapped by the user we record the change, recompute the `Set<Track>` of all selected tracks, save it in the `TracksFilter` and invoke `notifyItemChanged` on the adapter, so the item is rebound and the UI is always up to date.